pkg/cloud/scope/cluster.go

```diff
@@ -188,8 +188,8 @@ func (s *ClusterScope) ListOptionsLabelSelector() client.ListOption {
 	})
 }
 
-// Close closes the current scope persisting the cluster configuration and status.
-func (s *ClusterScope) Close() error {
+// Update persists the cluster configuration and status.
+func (s *ClusterScope) Update() error {
```

Comment:
Is the change to `Update()` necessary? Maybe add another method called `PatchObject` or similar?

Reply:
It's not a necessary change, but since I was using the `Close` function to update the AWSCluster with the finalizer, it didn't seem like the right name or the right description of what it was achieving.

Can revert if that makes the most sense. But probably makes sense to determine if the call to clusterScope.Close() is correct and necessary in `awscluster_controller.go` first.

Reply:
Close makes sense to be the last call, let's move the patching to a `PatchObject()` method, which you can call directly, and keep `Close` to be the last call. Close calls PatchObject.

Reply:
@vincepri I pushed an update. Let me know if I was able to capture what you were looking for in terms of creating a `PatchObject` method and having `Close` call `PatchObject`